src/components/Layout/Header/NavBar/KeepKey/ChangePin.tsx

```diff
@@ -1,50 +1,139 @@
-import { Button, Flex } from '@chakra-ui/react'
+import { Box, Button, Flex, useColorModeValue } from '@chakra-ui/react'
 import { useToast } from '@chakra-ui/toast'
 import { useTranslate } from 'react-polyglot'
+import { CircularProgress } from 'components/CircularProgress/CircularProgress'
 import { AwaitKeepKey } from 'components/Layout/Header/NavBar/KeepKey/AwaitKeepKey'
-import { LastDeviceInteractionStatus } from 'components/Layout/Header/NavBar/KeepKey/LastDeviceInteractionStatus'
 import { SubmenuHeader } from 'components/Layout/Header/NavBar/SubmenuHeader'
+import { Text } from 'components/Text'
+import { WalletActions } from 'context/WalletProvider/actions'
+import { KeepKeyPin } from 'context/WalletProvider/KeepKey/components/Pin'
+import { PinMatrixRequestType } from 'context/WalletProvider/KeepKey/KeepKeyTypes'
 import { useKeepKey } from 'context/WalletProvider/KeepKeyProvider'
 import { useWallet } from 'hooks/useWallet/useWallet'
 import { logger } from 'lib/logger'
 
+import { useMenuRoutes } from '../hooks/useMenuRoutes'
 import { SubMenuBody } from '../SubMenuBody'
 import { SubMenuContainer } from '../SubMenuContainer'
+import { LastDeviceInteractionStatus } from './LastDeviceInteractionStatus'
 
 const moduleLogger = logger.child({
   namespace: ['Layout', 'Header', 'NavBar', 'KeepKey', 'ChangePin'],
 })
 
 export const ChangePin = () => {
+  const { handleBackClick } = useMenuRoutes()
   const translate = useTranslate()
   const { keepKeyWallet } = useKeepKey()
   const {
+    dispatch,
     state: {
-      deviceState: { awaitingDeviceInteraction },
+      keepKeyPinRequestType,
+      deviceState: { awaitingDeviceInteraction, isUpdatingPin, isLoading },
     },
+    setDeviceState,
   } = useWallet()
   const toast = useToast()
+  const pinButtonsBackground = useColorModeValue('gray.200', 'gray.600')
+  const pinButtonsBackgroundHover = useColorModeValue('gray.100', 'gray.500')
+
+  const translationType = (() => {
+    switch (keepKeyPinRequestType) {
+      case PinMatrixRequestType.NEWFIRST:
+        return 'newPin'
+      case PinMatrixRequestType.NEWSECOND:
+        return 'newPinConfirm'
+      default:
+        return 'pin'
+    }
+  })()
+
+  const handleCancel = async () => {
+    const fnLogger = moduleLogger.child({ namespace: ['handleChangePinBackClick'] })
+
+    await keepKeyWallet?.cancel().catch(e => {
+      fnLogger.error(e, 'Error cancelling new PIN...')
+      toast({
+        title: translate('common.error'),
+        description: e?.message?.message ?? translate('common.somethingWentWrong'),
+        status: 'error',
+        isClosable: true,
+      })
+    })
+
+    setDeviceState({
+      isUpdatingPin: false,
+    })
+  }
+
+  const handleHeaderBackClick = async () => {
+    await handleCancel()
+
+    handleBackClick()
```

Comment:
We're missing an `await`.

```suggestion
    await handleBackClick()
```